frame/staking/src/lib.rs

```diff
@@ -1835,3 +2309,56 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 		}
 	}
 }
+
+#[allow(deprecated)]
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
```

Comment:
Just a huge note here:
Since we only check the signature here, validators are allowed to submit incorrect solutions (for free), and they will only fail when we try to apply them.
That should be ok, since it will be trivial to identify such malicious attempts and penalize validators on-chain manually.

Reply:
We discussed this offline and we had some ideas, but for now maybe we don't implement any. A new check is added that the score of an unsigned solution is checked in validate.